.github/workflows/admin-server.yml

```diff
@@ -0,0 +1,73 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Admin Panel Server CI/CD Pipeline
+
+on:
+  push:
+    paths:
+    - 'server/**'
+    # branches:
+    #   - master
+  pull_request:
+    paths:
+    - 'server/**'
+    branches:
+      - master
+
+env: 
+  project-directory: ./server
+
+jobs:
+  server: #Job ID Name
+    name: Build Server Project
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+    - name: npm clean install
+      run: npm ci
+      working-directory: ${{env.project-directory}}
+    - name: run ESLint
```

Comment:
Ideally, we want CI to run on all branches in case there are compilation failures and we can catch them earlier on rather than when we merge into master.

Reply:
Yep compilation and tests on branches makes sense to me